### PR TITLE
fix(facetValues): retrieve all hierarchical facet parent values

### PR DESCRIPTION
**Pull request by dhayab**
This PR retrieves all the values of parent hierarchical facet levels to display them, instead of only having this behavior for the root level and only showing the refined parent level value for the rest of the hierarchy.

Ticket: [FX-1458](https://algolia.atlassian.net/browse/FX-1458)

**Current implementation**

For a hierarchical refinement, when an nth-level value is refined, an additional query is generated to retrieve the ful llist of the root values, as by default only the refined value of each parent level will be returned.

It seems that this is by design (I might be wrong though, I based my observation on [this commit](https://github.com/algolia/algoliasearch-helper-js/commit/01cb48da841f2a4e740fe5cc5f870bbc4f48b00b)).

**New implementation**

Instead of only generating one query for the root level, we now generate as much queries as necessary in order to retrieve all the facet values of parent levels.

And when building the search results, we merge those values to get the full list.

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -632,27 +622,6 @@ algolia-frontend-components@0.0.35:
     webpack "^2.2.1"
     webpack-dev-server "^2.4.1"
 
-"algoliasearch-v3@npm:algoliasearch@3.35.1":
```
Comment (Haroenv) [inline yarn.lock:635]: maybe this should be done in a separate PR, but it was wrongly changed in https://github.com/algolia/algoliasearch-helper-js/pull/898

**Comment by Haroenv** [comment]
I might be missing something, but on https://codesandbox.io/s/instantsearch-js-forked-jebm3n?file=/src/app.js "Appliances > Dishwashers > Built-in Dishwashers" (https://of8n4x.csb.app/?instant_search%5BhierarchicalMenu%5D%5BhierarchicalCategories.lvl0%5D%5B0%5D=Appliances&instant_search%5BhierarchicalMenu%5D%5BhierarchicalCategories.lvl0%5D%5B1%5D=Dishwashers&instant_search%5BhierarchicalMenu%5D%5BhierarchicalCategories.lvl0%5D%5B2%5D=Built-In%20Dishwashers) still hides the results of lvl1

**Comment by dhayab** [comment]
@Haroenv could it because that's a transitive dependency? It works well locally when linked. I'll try to set up a working sandbox.

File: src/SearchResults/index.js [hunk 2]
```diff
@@ -533,47 +533,49 @@ function SearchResults(state, results, options) {
...
-      );
-    });
+        // when we always get root levels, if the hits refinement is `beers > IPA` (count: 5),
+        // then the disjunctive values will be `beers` (count: 100),
+        // but we do not want to display
+        //   | beers (100)
+        //     > IPA (5)
+        // We want
+        //   | beers (5)
+        //     > IPA (5)
```
[the remark this answers is not in the document]
Comment (dhayab) [reply src/SearchResults/index.js:562]: I agree, the count does not change in iOS for instance.

**Review by FabienMotte — APPROVED** [review]
